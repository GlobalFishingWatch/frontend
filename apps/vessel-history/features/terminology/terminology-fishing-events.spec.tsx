import { Trans, useTranslation } from 'react-i18next'
import { render } from '@testing-library/react'

import TerminologyFishingEvents from './terminology-fishing-events'

jest.mock('react-i18next')

describe('TerminologyFishingEvents', () => {
  const mockUseTranslation: jest.Mock = useTranslation as jest.Mock
  const mockTrans: jest.Mock = Trans as jest.Mock
  beforeEach(() => {
    mockUseTranslation.mockReturnValue({
      t: (_ns: any, fallback: any) => fallback,
    })
    mockTrans.mockImplementation(({ i18nKey, children }) => children)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render successfully', () => {
    const { baseElement } = render(<TerminologyFishingEvents />)
    expect(baseElement).toMatchSnapshot()
  })
})
