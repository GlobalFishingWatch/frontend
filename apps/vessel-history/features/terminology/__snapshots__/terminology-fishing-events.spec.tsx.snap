// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminologyFishingEvents should render successfully 1`] = `
<body>
  <div>
    <p>
      When there is a consistent series of AIS points estimated as indicating apparent fishing activity the points are grouped into 'Fishing Events'. 
      <br />
      We analyze the automatic identification system (AIS) data collected from vessels that our research has identified as known or possible commercial fishing vessels, and apply a fishing detection algorithm to determine “apparent fishing activity” based on changes in vessel speed and direction.
      <br />
      The algorithm classifies each AIS broadcast data point for these vessels as either “apparently fishing” or “not fishing” and shows the former on our fishing activity heat map. You can find more detail in our Technology page:
       
      <a
        href="https://globalfishingwatch.org/our-technology/"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://globalfishingwatch.org/our-technology/
      </a>
       
      and Dataset page:
       
      <a
        href="https://globalfishingwatch.org/datasets-and-code/"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://globalfishingwatch.org/datasets-and-code/
      </a>
      .
    </p>
  </div>
</body>
`;
